onnxruntime/core/providers/migraphx/migraphx_allocator.cc

```diff
@@ -51,27 +51,27 @@ void* MIGraphXExternalAllocator::Alloc(size_t size) {
 
 void MIGraphXExternalAllocator::Free(void* p) {
   free_(p);
-  std::lock_guard<std::mutex> lock(lock_);
-  auto it = reserved_.find(p);
-  if (it != reserved_.end()) {
+  std::lock_guard lock(lock_);
```

Comment:
Why remove the template type here? What did Clang-Tidy suggest here? It doesn't make sense to make this less explicit and rely on the default template type which is only valid in C++17, is that what we want here?